tools/tc39-test262/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+//! TC39 test262
+mod edition;
+mod structs;
+mod test_files;
+mod test_flags;
+
+pub mod git;
+pub mod read;
+pub use structs::{ErrorType, Outcome, Phase, Ignored};
+pub use test_files::{Harness, HarnessFile, MetaData, Test, TestSuite};
+pub use test_flags::TestFlags;
+pub use edition::SpecEdition;
+
+/// Repository Url
+pub const TEST262_REPOSITORY: &str = "https://github.com/tc39/test262";
+/// Git clone directory
+pub const TEST262_DIRECTORY: &str = "test262";
+
+/// Clone TC39 test262 repostiory
+pub fn clone_test262(commit: Option<&str>, verbose: u8) -> color_eyre::Result<()> {
+    const TEST262_REPOSITORY: &str = "https://github.com/tc39/test262";
+    git::clone(
+        TEST262_DIRECTORY,
+        TEST262_REPOSITORY,
+        &"origin/main",
+        commit,
+        verbose,
+    )
+}
```

Comment:
I'm not sure if we should expose a `clone_repo` function. IMO, we should provide a `test262` function that takes a struct of options and returns the whole suite, optionally cloning the repo if a path to a valid repo is not provided.

Reply:
After longer consideration and refactor I not sure if reading and optionally cloning logic should be in the same function (double responsibility).  

- it would add more complexity to read(...) fn signature as we would have to provide extra information (struct) on each read level (harness, suite, test).
- if we provide it only on the suite level api in that form can be confusing and inconsistent. 

Usage in main: 

![image](https://github.com/user-attachments/assets/d1929ffc-fe2f-4b0e-847a-7ce862fbc00a)

> in this example if path to suite would be a direct path to test (.js) test262 would skip clone and return error, causing inconsistent cli behavior.
 
 @jedel1043 please let me know what do you think about it. Maybe I'm missing something and there is a better solution to make it done.

Reply:
@jedel1043  Thanks for clarification. 

Please take a look at this changes and let me know if there are closer to the expected result.
https://github.com/boa-dev/boa/pull/3708/commits/027fbb4f99d98cae5a9ec96a33ac542a181f8051?diff=split&w=0

Still some work is needed, but I want to make sure that this is the desired direction.